src/Lean/Elab/SyntheticMVars.lean

```diff
@@ -208,6 +208,94 @@ private def synthesizeUsingDefault : TermElabM Bool := do
       return true
   return false
 
+/-- Translate zero-based indexes (0, 1, 2, ...) to ordinals ("first", "second", "third", ...)-/
+private def toOrdinalString : Nat -> String
+  | 0 => "first"
+  | 1 => "second"
+  | 2 => "third"
+  | 3 => "fourth"
+  | 4 => "fifth"
+  | n => s!"{n+1}th"
+
+/-- Make an oxford-comma-separated list of strings. -/
+private def toOxford : List String -> String
+  | [] => ""
+  | [a] => a
+  | [a, b] => a ++ " and " ++ b
+  | [a, b, c] => a ++ ", " ++ b  ++ ", and " ++ c
+  | a :: as => a ++ ", " ++ toOxford as
+
+/- Give alternative forms of a string if the `count` is 1 or not. -/
+private def _root_.Nat.plural (count : Nat) (singular : String) (plural : String) :=
+  if count = 1 then
+    singular
+  else
+    plural
```

Comment:
I don't think we want this as part of the public-facing Lean API, but these are tools that can be useful for writing good error messages. @Kha , two questions I think for you:

- Do you have thoughts on where in the lean4 tree I could collect some helper functions along these lines?
- Am I correct that I can do this nat-extension game without polluting the global `Nat` namespace with module system privacy? It's easy enough to have a `plural n sing pl` function but `n.plural sing pl` felt like it read nicely.